### ISSUE TITLE
Quarkus (gizmo): support @ShadowVariable on field too

### ISSUE THREAD

**Issue by ge0ffrey** [TimefoldAI/timefold-solver#7]
To reproduce:

1. Open timefold-quickstarts, food-packaging, Job
2. Switch @ShadowVariable and @PiggybackShadowVariable from getter to field annotations
3. Run food-packaging quickstart

```
 Failed to start application (with profile [dev]): java.lang.IllegalStateException: Member (startCleaningDateTime) of class (org.acme.foodpackaging.domain.Job) is not public and domainAccessType is GIZMO.
Maybe put the annotations onto the public getter of the field.
Maybe use domainAccessType REFLECTION instead of GIZMO.
        at ai.timefold.solver.core.impl.domain.common.accessor.gizmo.GizmoMemberDescriptor.<init>(GizmoMemberDescriptor.java:49)
        at ai.timefold.solver.core.impl.domain.common.accessor.gizmo.GizmoMemberAccessorImplementor.createAccessorFor(GizmoMemberAccessorImplementor.java:107)
        at ai.timefold.solver.core.impl.domain.common.accessor.gizmo.GizmoMemberAccessorFactory.buildGizmoMemberAccessor(GizmoMemberAccessorFactory.java:40)
```

**Comment by ge0ffrey** [comment]
Related to https://github.com/TimefoldAI/timefold-solver/issues/117 but different

**Comment by ge0ffrey** [comment]
the ofr model still suffers from this

**Comment by Christopher-Chianelli** [comment]
Fixed by #412
